### PR TITLE
[4737] Support: add autocomplete to search for users

### PR DESCRIPTION
**Pull request by jhenderson**
### Context
https://trello.com/c/PbsihW8T/4737-m-support-add-autocomplete-to-search-for-users

### Changes proposed in this pull request
- Add autocomplete functionality to user search input field on `/system-admin/users`

### Guidance to review
- Login as system admin
- Visit the users page
- Begin search for users
- Suggestions should appear
- Choose one
- Page should automatically redirect to user page

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by jhenderson** [comment]
[[M] Support: add autocomplete to search for users](https://trello.com/c/PbsihW8T/4737-m-support-add-autocomplete-to-search-for-users)

File: config/analytics.yml [hunk 1]
```diff
@@ -268,6 +268,7 @@ shared:
     - system_admin
     - updated_at
     - welcome_email_sent_at
+    - searchable
```
Comment (AbigailMcP) [inline config/analytics.yml:271]: I'm not sure this is useful for sending to bigquery - i see we added the school `searchable` to the blocklist.

File: app/services/user_search.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  def call
+    Result.new(users: specified_users, limit: limit)
+  end
+
+  def specified_users
+    users = User.all
+    users = users.search(query) if query
+    users = users.limit(limit) if limit
+    users
```
Comment (AbigailMcP) [inline app/services/user_search.rb:31]: Hmmm I wonder whether we can apply some kind of sort here, or what it feels like without a sort with out full dataset..

**Review by AbigailMcP — COMMENTED** [review]
Looks good! A couple of things i noticed:

### 1. Diff between search results and dropdown results
If I search for "bell", only Annie Bell shows up in the dropdown:
Screenshot 2022-12-13 at 15 12 34

However, if I then click 'Search':

Screenshot 2022-12-13 at 15 12 44

A wild Damian appears. I think Damian should be being returned in the dropdown.

### 2. No obvious way to 'clear search'
Not sure whether this is a big deal because it's an admin feature, maybe one for @edwardhorsford , but after I've searched for someone, there's no way to clear the search other than using browser back or re-clicking the 'Users' tab.

Screenshot 2022-12-13 at 15 16 04

Related: the field doesn't 'remember' the search term, but again not sure whether that's a biggie.

### 3. Form heading

I think 'Search for a user' would probably look better bold as in the prototype.

### 4. Field hint

The prototype says "Search by name, email or organisation"

**Review by edwardhorsford — APPROVED** [review]

**Review by AbigailMcP — APPROVED** [review]
looks nice!

i'd kinda expect using keyboard presses 'down' and 'enter' on a name to redirect to the user page rather than submit the form (i.e. the same as clicking on a person with the mouse)

but i like the clear search and how we're showing the search term now 👍

**Comment by jhenderson** [comment]
> looks nice!
> 
> i'd kinda expect using keyboard presses 'down' and 'enter' on a name to redirect to the user page rather than submit the form (i.e. the same as clicking on a person with the mouse)
> 
> but i like the clear search and how we're showing the search term now 👍

@AbigailMcP Selecting a user should redirect to the user's page. My recent update introduced a regression. I've since fixed that.

**Comment by AbigailMcP** [comment]
> > looks nice!
> > i'd kinda expect using keyboard presses 'down' and 'enter' on a name to redirect to the user page rather than submit the form (i.e. the same as clicking on a person with the mouse)
> > but i like the clear search and how we're showing the search term now 👍
> 
> @AbigailMcP Selecting a user should redirect to the user's page. My recent update introduced a regression. I've since fixed that.

@jhenderson nice! tested and works 💯

**Pull request merged**